README.md

```diff
@@ -0,0 +1,116 @@
+and [Microsoft Accounts](https://account.microsoft.com) in a converged experience using industry standard OAuth2 and OpenID Connect.
+Soon MSAL Python will also support [Azure AD B2C](https://azure.microsoft.com/services/active-directory-b2c/).
+
+More and more detail about MSAL Python functionality and usage will be documented in the
+[Wiki](https://github.com/AzureAD/microsoft-authentication-library-for-python/wiki).
+
+## Installation
+
+1. If you haven't already, [install and/or upgrade the pip](https://pip.pypa.io/en/stable/installing/)
+   of your Python environment to a recent version. We tested with pip 18.1.
+2. For now, you can install from our latest dev branch, by `pip install https://github.com/AzureAD/microsoft-authentication-library-for-python.git@dev`
+
+## Usage
+
+Before using MSAL Python (or any MSAL SDKs, for that matter), you will have to
+[register your application with the AAD 2.0 endpoint](https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-v2-register-an-app).
+
+Acquiring tokens with MSAL Python is somewhat different than ADAL Python. You will need to follow this 3-step pattern.
+
+1. Contrary to ADAL (which proposes the notion of `AuthenticationContext`, which is a connection to Azure AD),
+   MSAL proposes a clean separation between
+   [public client applications, and confidential client applications](https://tools.ietf.org/html/rfc6749#section-2.1).
+   So you will first create either a `PublicClientApplication` or a `ConfidentialClientApplication` instance,
+   and ideally reuse it during the lifecycle of your app. For example:
+
+   ```python
+   from msal import PublicClientApplication
+   app = PublicClientApplication("your_client_id", authority="...")
+   ```
+   
+   Later, each time you would want an access token, you start by:
+   ```python
+   result = None
+   ```
+
+2. The API model in MSAL provides you explicit control on how to utilize token cache.
+   This cache part is technically optional, but we highly recommend you to harness the power of MSAL cache.
+   It will automatically handle the token refresh for you.
+
+   ```python
+   # We now check the cache to see if we have some end users already signed in before.
+   accounts = app.get_accounts()
+   if accounts:
+       # If so, you could then somehow display these accounts and let end user choose
+       print("Pick the account you want to use to proceed:")
+       for a in accounts:
+           print(a["username"])
+       # Assuming the end user chose this one
+       chosen = accounts[0]
+       # Now let's try to find a token in cache for this account
+       result = app.acquire_token_silent(config["scope"], account=chosen)
+   ```
+
+3. Either there is no suitable token in the cache, or you chose to skip the previous step,
+   now it is time to actually send a request to AAD to obtain a token.
+   There are different methods based on your client type. Here we demonstrate the username password flow.
```

Comment:
Can we demonstrate auth code or client creds flow? Since we don't want to encourage username password flow, I think it is better to not have it as the Readme example.

Reply:
I understand and I did consider this rationale while writing this first draft. The trade-off here is, this (short?) README can only cover that much. So I think we would like to quickly mention an overall usage pattern of MSAL, i.e. the step 1-2-3 thing. In that sense, the Username Password flow is conceptually easiest to NOT get in the way. On the contrary, the auth code flow happens to be too complicated when it incorporates web service framework/logic.

On a side note, I/we do plan to work on an auth code sample, which is the only missing part so far. Later we will have dedicated links to those samples, and we will put auth code sample link here in the top place, and also somehow document that we do not recommend U/P flow, perhaps in its sample too.

Reply:
Details should be in the wiki not in the readme, imo.

Reply:
I agree that the details should be in the Wiki and I'm not suggesting covering all the flows here.
The Readme is the first page people see when they look at the repo and I think it should show an acquireToken pattern which is more widely recommended(than the U/P flow) even if a bit more complicated instead of choosing by ease of the pattern. And since we don't yet have auth code flow, I suggest the client credential flow with a client secret which is also simple.